### ISSUE TITLE
KDE Global Menu get messed up

### ISSUE THREAD

**Issue by Icenowy** [muse-sequencer/muse#1292]
**Describe the bug**
When running MusE under KDE, the global menu could not be displayed properly, and lacks many functionalities.

**To Reproduce**
- Use KDE and set up a Global Menu Plasma applet. (I use KDE 5 here).
- Start MusE and Alt-Tab to another program, then switch back.

**Expected behavior**
The global menu should match the menu that ought to be shown in MusE.

**Screenshots**
![image](https://github.com/user-attachments/assets/438c39a8-e6c0-4e59-b43b-48811ab4b5e1)

**Desktop (please complete the following information):**
 - OS: AOSC OS
 - Desktop: KDE 5

```
Operating System: AOSC OS 11.6.1
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.13
Kernel Version: 6.11.4-aosc-main (64-bit)
Graphics Platform: X11
```

**Additional context**
I tried to compile MusE by myself (because packaging is still lacking on AOSC OS) and to use AppImage, the same issue persists. With compiled MusE it recognizes `KDE_NO_GLOBAL_MENU=1` environment variable and is able to workaround this problem, but with AppImage even this workaround is not available.

**Comment by terminator356** [comment]
Thanks for the report.
Wow. I had no idea such an applet existed all these years.
Tested on OpenSUSE Tumbleweed (rolling distro, latest daily changes): 
Same problem observed.

Bizarre. I read that the applet is not well supported and there is talk of removing it.

It probably does not help that MusE's menu system is not 'common' usage and rather 
 customized behaviour in some places, such as dynamically modifying menus 'on the fly'.

However I am surprised that not even the Global Settings menu appears, for example.

This is a clue that the applet indeed does not like our menu system, because some of our menus
 are put there by our main window, and some are put there by our 'Arranger' widget, 
 and some are put there by our 'ArrangerView' widget. As well, the editor windows put their own
 menus. All dynamically depending on which window is being viewed. 
In other words, we are sort of 'counteracting' what the applet is trying to do, because we're
 sort of already doing it ourselves.

Because of that, if the clue is correct, unfortunately at this time I could not recommend using MusE with the applet.
It is likely that attempting to fix the problem would be overwhelming and not worth the effort.

Unless... If someone could point out a Qt (not KDE) compilation flag that I could turn on the the official release,
 to disable the global menu usage?

**Comment by terminator356** [comment]
Further investigation: Someone added a MusE command-line option, some years ago:

-u   Ubuntu/unity workaround: don't allow sharing menus and mdi-subwins.

Unfortunately this means it forces all the windows to be separate (and thus each has its own 'static' menu).
You would lose MusE's nice tabbed windows feature.

But... even worse, the option seems partially broken at the moment.
The Arranger window refuses to appear - unless you load a song, **then it is fine**.
I tried everything to make it appear at startup. No luck.

This might be some relatively simple oversight or omission in the code, since we can see 
that the Arranger _can_ in fact appear, just not at startup.

But... Even if the -u option is fixed, the many separate windows instead of the tabbed windows 
makes it likely to be awkward.

Some history: Separate windows is how MusE used to look a long time ago.
Then, we added three UI style choices: 
1) Separate windows, each with its own menu.
2) Multiple Document Interface (MDI) style (all windows inside one container window).
3) Mac/Borland style. Separate windows and one shared menu for all windows, 
like Mac or Borland Delphi/C++ Builder. (It was more like Borland since it was 
not a truly global menu, it was only for MusE.)

Then, a few years ago, the tabbed window feature was added.
Users can still separate each window if they wish.

As you can see, we have a history of complex menu systems with dynamic on-the-fly changes!

I will see about contacting the author to see if there is any hope of MusE cooperating with it.

**Comment by Icenowy** [comment]
For AOSC OS, I am going to add a qputenv() in main(), but don't know whether this will be helpful to AppImage build... (I assume not because running the AppImage with the env var fails)

**Comment by Icenowy** [comment]
In addition, maybe I should file a bug report to Plasma?

**Comment by terminator356** [comment]
> I am going to add a qputenv() in main()

Yes! I was just going to do that myself. Although I cannot test the AppImage.
Tomorrow I'll do it.

> bug report to Plasma?

Yes. Done! It may take some time for a response.

**Comment by Icenowy** [comment]
To be honest, I have no idea why the AppImage uses the global menu. Global menu integration seems to be a plugin into the Qt of my host, but the AppImage is running with its own Qt, and that Qt has no KDEPlasmaPlatformTheme.so module...

**Comment by Icenowy** [comment]
I read the Qt document and source code, and found that Qt itself have support for Ubuntu's dbusmenu protocol (which KDE tries to keep compatible with).

In addition, I found an application attribute named AA_DontUseNativeMenuBar, which disables global menu on Linux/Mac. Adding a `app.instance()->setAttribute(Qt::AA_DontUseNativeMenuBar, true);` in main.cpp seems to allow the application to force the use of in-window menu.

**Comment by terminator356** [comment]
I saw that flag last night but had not investigated yet.
Thanks very much for the work. You are now immortalized in the ChangeLog.

It seems this flag might fix Unity and Mac as well.
See QMenuBar help: "QMenuBar as a Global Menu Bar".
It specifically mentions MacOs and Ubuntu Unity.

So I am wondering if our -u command-line option was necessary.
Maybe the author wasn't aware of the flag or it didn't work.
I see no mention in the ChangeLog or code.

Cheers.
